trigger.go

```diff
@@ -0,0 +1,99 @@
+package cosmosdb
+
+import (
+	"context"
+)
+
+type Trigger struct {
+	Resource
+	Id        string           `json:"id"`
+	Body      string           `json:"body"`
+	Operation TriggerOperation `json:"triggerOperation"`
+	Type      TriggerType      `json:"triggerType"`
+}
+
+type TriggerType string
+type TriggerOperation string
+
+type CollectionTriggers struct {
+	Rid      string    `json:"_rid,omitempty"`
+	Count    int32     `json:"_count,omitempty"`
+	Triggers []Trigger `json:"Triggers"`
+}
+
+//const (
+//	TriggerTypePost = TriggerType("Post")
+//	TriggerTypePre  = TriggerType("Pre")
+//
+//	TriggerOpAll     = TriggerType("All")
+//	TriggerOpCreate  = TriggerType("Create")
+//	TriggerOpReplace = TriggerType("Replace")
+//	TriggerOpDelete  = TriggerType("Delete")
+//)
+
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-trigger
+type TriggerCreateOptions struct {
```

Comment:
Same as for the collections, would be good to reserve `Options` for the headers. It is identical with `Trigger` anyway, except of the resource. But same reasoning as before as well, the expected request body is well known, so maybe there's a point in bundling it.